frontend/packages/knative-plugin/src/utils/create-knative-utils.ts

```diff
@@ -111,6 +111,7 @@ export const knativeServingResourcesRevision = (namespace: string): FirehoseReso
       kind: referenceForModel(RevisionModel),
       namespace,
       prop: 'revisions',
+      optional: true,
```

Comment:
Do you no longer see the problem switching projects? https://github.com/openshift/console/pull/2862#issuecomment-536995960

Reply:
yes it happens only in topology if we switch namespace and it's not associated with this change `optional:true`. I  mean even in current master it will be observed in case non-admin user tries to view projects not created by them and on the first load will see "Restricted access"  and on change of namespace will show loading if  knative serverless is installed and as we are showing knative specific resources in sidebar as well so in workloads user will be shown with "restricted access". As you suggested `optional: true` is working as expected here.